fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,410 @@
+	else:
+		userLayout.prop(userProp, "customActor")
+		if userProp.customActor:
+			prop_split(userLayout, userProp.actor, "actorIDCustom", "Actor ID Custom")
+
+		roomObj = getRoomObj(userObj)
+		if roomObj is not None:
+			split = userLayout.split(factor = 0.5)
+			split.label(text = "Room Index")
+			split.label(text = str(roomObj.ootRoomHeader.roomIndex))
+		else:
+			userLayout.label(text = "This must be part of a Room empty's hierarchy.", icon = 'OUTLINER')
+
+		prop_split(userLayout, userProp, "spawnIndex", "Spawn Index")
+		entranceBool = userProp.customActor
+
+	if (user != userEntrance and userActorID != 'Custom') or (user == userEntrance and entranceBool is False):
+		# If the current actor isn't custom
+		if user != userEntrance:
+			userLayout.label(text = currentActor)
+			typeText = 'Type'
+		else:
+			typeText = 'Spawn Type'
+
+		propAnnot = getattr(detailedProp, dpKey + '.type', None)
+		if propAnnot is not None:
+			prop_split(userLayout, detailedProp, dpKey + '.type', typeText)
+
+		if user == userActor:
+			rotXBool = rotYBool = rotZBool = False
+			if userActorID == detailedProp.actorID:
+				for actorNode in root:
+					if actorNode.get('ID') == userActorID:
+						for elem in actorNode:
+							if elem.tag == 'ChestContent':
+								# Draw chest content searchbox
+								searchOp = userLayout.operator(OOT_SearchChestContentEnumOperator.bl_idname, icon='VIEWZOOM')
+								drawOperatorBox(userLayout, userObj, detailedProp, 'itemChest', 'Chest Content', searchOp, ootChestContent)
+							if elem.tag == 'Message':
+								# Draw Navi message ID searchbox
+								searchOp = userLayout.operator(OOT_SearchNaviMsgIDEnumOperator.bl_idname, icon='VIEWZOOM')
+								drawOperatorBox(userLayout, userObj, detailedProp, 'naviMsgID', 'Message ID', searchOp, ootNaviMsgID)
+
+			propAnnot = getattr(detailedProp, dpKey + ('.collectibleDrop'), None)
+			if propAnnot is not None:
+				prop_split(userLayout, detailedProp, dpKey + '.collectibleDrop', 'Collectible Drop')
+
+		if user == userActor:
+			drawParams(userLayout, detailedProp, dpKey, dpKey + '.chestFlag', 'Chest Flag', 'Flag', 'Chest', None)
+			drawParams(userLayout, detailedProp, dpKey, dpKey + '.collectibleFlag', 'Collectible Flag', 'Flag', 'Collectible', None)
+
+		drawParams(userLayout, detailedProp, dpKey, f'{dpKey}.switchFlag', 'Switch Flag', 'Flag', 'Switch', getMaxElemIndex(dpKey, 'Flag', 'Switch'))
+		drawParams(userLayout, detailedProp, dpKey, f'{dpKey}.enum', None, 'Enum', None, None)
+		drawParams(userLayout, detailedProp, dpKey, f'{dpKey}.props', None, 'Property', None, None)
+		drawParams(userLayout, detailedProp, dpKey, f'{dpKey}.bool', None, 'Bool', None, None)
+
+		paramBox = userLayout.box()
+		paramBox.label(text="Actor Parameter")
+		params = getattr(detailedProp, userParamField, "")
+		if params != "": paramBox.prop(detailedProp, userParamField, text="")
+		else: paramBox.label(text="That Actor doesn't have parameters.")
```

Comment:
```suggestion
		else: paramBox.label(text="This Actor doesn't have parameters.")
```
what happens with custom actors btw?

Reply:
this code will execute if it's not a custom actor so it should be fine